### PR TITLE
[SD-863] Fixes 500 error when creating stock location with propagate all variants option

### PR DESCRIPTION
**Pull request by aplegatt**

### CODE REVIEW

File: core/app/models/spree/stock_location.rb [hunk 1]
```diff
@@ -106,9 +106,16 @@ def fill_status(variant, quantity)
     private
 
     def create_stock_items
-      Variant.includes(:product).find_each do |variant|
-        propagate_variant(variant)
+      prepared_stock_items = Spree::Variant.all.map do |variant|
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:109]:
```suggestion
      prepared_stock_items = Spree::Variant.ids.each do |variant_id|
```

File: core/app/models/spree/stock_location.rb [hunk 2]
```diff
@@ -106,9 +106,16 @@ def fill_status(variant, quantity)
...
+        Hash[
+          'stock_location_id', id,
+          'variant_id', variant.id,
+          'backorderable', true,
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:113]: @aplegatt this should be grabbed from stock_location config

File: core/app/models/spree/stock_location.rb [hunk 3]
```diff
@@ -106,9 +106,16 @@ def fill_status(variant, quantity)
     private
 
     def create_stock_items
-      Variant.includes(:product).find_each do |variant|
-        propagate_variant(variant)
+      prepared_stock_items = Spree::Variant.all.map do |variant|
+        Hash[
+          'stock_location_id', id,
+          'variant_id', variant.id,
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:112]:
```suggestion
          'variant_id', variant_id,
```

**Review by damianlegawiec — COMMENTED** [review]
@aplegatt good direction! 👍

File: core/spec/models/spree/stock_item_spec.rb [hunk 4]
```diff
@@ -246,14 +246,6 @@
           end.not_to change { subject.variant.reload.updated_at }
         end
       end
-
-      context 'when a new stock location is added' do
-        it 'touches its variant' do
```
Comment (aplegatt) [inline core/spec/models/spree/stock_item_spec.rb:251]: Not sure if we need variant to be touched when creating a new stock location?

Comment (damianlegawiec) [reply core/spec/models/spree/stock_item_spec.rb:251]: @aplegatt we can run touch for all Variants at once via single SQL query:

```
Spree::Variant.touch_all
```

https://apidock.com/rails/ActiveRecord/Relation/touch_all

Comment (damianlegawiec) [reply core/spec/models/spree/stock_item_spec.rb:251]: And yes we need it to invalidate cache

File: core/app/models/spree/stock_location.rb [hunk 5]
```diff
@@ -106,8 +106,18 @@ def fill_status(variant, quantity)
...
+          'stock_location_id', id,
+          'variant_id', variant_id,
+          'backorderable', backorderable_default,
+          'created_at', DateTime.current,
+          'updated_at', DateTime.current
+        ]
+      end
+      if prepared_stock_items.any?
+        stock_items.insert_all(prepared_stock_items)
+        Spree::Variant.touch_all
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:120]:
```suggestion
        variants_scope.touch_all
```

File: core/app/models/spree/stock_location.rb [hunk 6]
```diff
@@ -106,8 +106,18 @@ def fill_status(variant, quantity)
...
     def create_stock_items
-      Variant.includes(:product).find_each do |variant|
-        propagate_variant(variant)
+      prepared_stock_items = Spree::Variant.ids.map do |variant_id|
+        Hash[
+          'stock_location_id', id,
+          'variant_id', variant_id,
+          'backorderable', backorderable_default,
+          'created_at', DateTime.current,
+          'updated_at', DateTime.current
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:115]:
```suggestion
          'created_at', Time.current,
          'updated_at', Time.current
```

File: core/app/models/spree/stock_location.rb (hunk 5 above)
Comment (aplegatt) [reply core/app/models/spree/stock_location.rb:120]: This doesn't seem to work. I don't think we have such scope in StockLocation.

File: core/app/models/spree/stock_location.rb [hunk 7]
```diff
@@ -106,8 +106,18 @@ def fill_status(variant, quantity)
     private
 
     def create_stock_items
-      Variant.includes(:product).find_each do |variant|
-        propagate_variant(variant)
+      prepared_stock_items = Spree::Variant.ids.map do |variant_id|
```
Comment (damianlegawiec) [inline core/app/models/spree/stock_location.rb:109]:
```suggestion
      variants_scope = Spree::Variant
      prepared_stock_items = Spree::Variant.ids.map do |variant_id|
```

File: core/app/models/spree/stock_location.rb (hunk 5 above)
Comment (damianlegawiec) [reply core/app/models/spree/stock_location.rb:120]: @aplegatt sorry suggestion didn't add earlier - let's add it here: https://github.com/spree/spree/pull/10333#pullrequestreview-445748671

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**